src/transforms/util/pullRequestLinkGenerator.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Generates a create pull request link for GH in the format [URL]/compare/[name]?title=[...keys]-[name]
+ * This format allows the use of title query param to set the PR name
+ */
+export const generateCreatePullRequestUrl = (baseUrl: string, name: string, issueKeys?: string[]) => {
+	// Remove duplicate keys
+	const uniqueIssueKeys: string[] = [...new Set(issueKeys)];
+	let title: string | undefined = uniqueIssueKeys?.join(" ");
+
+	// Only prepend issue keys if they exist
+	if (title) {
+		title = encodeURIComponent(`${title} - ${name}`);
+	} else {
+		title = encodeURIComponent(`${name}`);
+	}
+	return  `${baseUrl}/compare/${name}?title=${title}`;
```

Comment:
You'll need to `encodeURIComponent` of name here.

An idea I had for this since you're doing url params is you could use the `URLParamsMiddleware` function to do what you want:

```
return urlParamsMiddleware({
   url: "{baseUrl}/compare/{name}?title={title}"
   urlParams: {
      baseUrl,
      name,
      title
   }
}).url;
```

Might be overkill though.  I think we might need to create a shared util from the logic in URLParamsMiddleware that does this for any url.

Reply:
After the changes from your previous comments the function is quite succinct, so will omit the urlParamsMiddleware at this stage.

Reply:
You still need to `encodeURIComponent` of name since a branch name can have `/` in it.

Reply:
also, don't you need `quick_pull=1` in the querystring to make it a PR right away?  because the first page here is just a compare.  By adding this in the query, you go directly to the create PR page.

Reply:
I have the return encoding the entire string 
	return  ``${baseUrl}/compare/${name}?title=${title}`;` that should cover it?

Reply:
appended quick_pull=1 too, can confirm it goes direct to PR.

Reply:
But if we encode the entire string we also encode `/` and '?' so the URL will become invalid.

We need to encode each parameter individually. (The reason to encode the string to prevent injection attacks, so nobody will be able to put special characters in URL and make it pointing to another path)

Reply:
So it should be
`${ encodeURIComponent(baseUrl)}/compare/${ encodeURIComponent(name)}?title=${ encodeURIComponent(title)}`

Reply:
I accept, i have encoded each individually

	const title = encodeURIComponent(keys + name);
	const branchName = encodeURIComponent(name);
	const base = encodeURIComponent(baseUrl);
	const url = `${base}/compare/${branchName}?title=${title}&quick_pull=1`